shepherd_bms/src/stateMachine.cpp

```diff
@@ -213,177 +193,84 @@ void StateMachine::requestTransition(BMSState_t next_state)
 }
 
 
-uint32_t StateMachine::faultCheck(AccumulatorData_t *accData)
+uint32_t StateMachine::faultReturn(AccumulatorData_t *accData)
 {
 	// FAULT CHECK (Check for fuckies)
-	uint32_t faultStatus = 0;
 
-	//prefault for Low Cell Voltage
-
-	if (prefaultLowCell.faultEvalState == BEFORE_TIMER_START && accData->min_voltage.val < MIN_VOLT * 10000)
-    {
-        prefaultLowCell.faultTimer.startTimer(PRE_UNDER_VOLT_TIME);
-        prefaultLowCell.faultEvalState = DURING_FAULT_EVAL;
-    }
-    else if (prefaultLowCell.faultEvalState == DURING_FAULT_EVAL)
-    {
-        if (prefaultLowCell.faultTimer.isTimerExpired())
-        {
-            if (prefaultCANDelay1.isTimerExpired())
-			{
-            	compute.sendDclPreFault(true);
-				prefaultCANDelay1.startTimer(CAN_MESSAGE_WAIT);
-			}
-        }
-        if (!(accData->min_voltage.val < MIN_VOLT * 10000))
+	struct fault_eval fault_table[8] = 
         {
-            prefaultLowCell.faultTimer.cancelTimer();
-            prefaultLowCell.faultEvalState = BEFORE_TIMER_START;
-        }
-    }
+          // ___________FAULT ID____________   __________TIMER___________   _____________DATA________________    __OPERATOR__   __________________________THRESHOLD____________________________  _____________FAULT CODE_________________    _______________DATA_________________  ___OPERATOR___  ________THRESHOLD________
```

Comment:
did u view in github or vscode? i really wasnt sure how to make it better lol but i agree its overwhelming, if you have specific ideas lmk otherwise i can keep messing with it